frontend/src/main/components/RequestType/RequestTypeTable.js

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import OurTable, { ButtonColumn } from "main/components/OurTable";
+
+import { useBackendMutation } from "main/utils/useBackend";
+import {
+  cellToAxiosParamsDelete,
+  onDeleteSuccess,
+} from "main/utils/RequestTypeUtils";
+import { useNavigate } from "react-router-dom";
+import { hasRole } from "main/utils/currentUser";
+
+export default function RequestTypeTable({ requesttype, currentUser }) {
+  const navigate = useNavigate();
+
+  const editCallback = (cell) => {
+    navigate(`/requesttypes/create/${cell.row.values.id}`);
+  };
+
+  // Stryker disable all : hard to test for query caching
+
+  // when delete success, invalidate the correct query key (depending on user role)
+  const apiEndpoint = hasRole(currentUser, "ROLE_PROFESSOR")
+    ? "/api/requesttypes/all"
+    : "/api/requesttypes/all";
+
+  const deleteMutation = useBackendMutation(
+    cellToAxiosParamsDelete,
+    { onSuccess: onDeleteSuccess },
+    [apiEndpoint],
+  );
+  // Stryker restore all
+
+  // Stryker disable next-line all : TODO try to make a good test for this
+  const deleteCallback = async (cell) => {
+    deleteMutation.mutate(cell);
+  };
+
+  const columns = [
+    {
+      Header: "id",
+      accessor: "id", // accessor is the "key" in the data
+    },
+    {
+      Header: "Request Type",
+      accessor: "requestType",
+    },
+  ];
+
+  //since all admins have the role of a user, we can just check if the current user has the role ROLE_USER
+  if (hasRole(currentUser, "ROLE_USER")) {
+    columns.push(
+      ButtonColumn("Delete", "danger", deleteCallback, "RequestTypeTable"),
+    );
+  }
+
+  if (
+    hasRole(currentUser, "ROLE_USER") &&
+    (hasRole(currentUser, "ROLE_ADMIN") ||
+      hasRole(currentUser, "ROLE_PROFESSOR"))
+  ) {
+    columns.push(
+      ButtonColumn("Edit", "primary", editCallback, "RequestTypeTable"),
+    );
+  }
```

Comment:
Here, there is no need to check for ROLE_USER.   If someone has ROLE_ADMIN or ROLE_PROFESSOR, they automatically have ROLE_USER.
```suggestion

  if (  hasRole(currentUser, "ROLE_ADMIN") ||
        hasRole(currentUser, "ROLE_PROFESSOR")
  ) {
    columns.push(
      ButtonColumn("Edit", "primary", editCallback, "RequestTypeTable"),
    );
  }
```

Reply:
And checking for ROLE_ADMIN or ROLE_PROFESSOR is probably what you needed for delete too, I'm guessing.

It definitely doesn't make sense for ROLE_USER to have the delete button, but only admins and professors to be able to edit.  So something is off here.